spark/dl/src/main/scala/org/apache/spark/ml/DLClassifier.scala

```diff
@@ -15,147 +15,74 @@
  */
 package org.apache.spark.ml
 
-import scala.reflect.ClassTag
-
-import com.intel.analytics.bigdl.Module
-import com.intel.analytics.bigdl.models.utils.ModelBroadcast
+import com.intel.analytics.bigdl.tensor.Tensor
 import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
-import com.intel.analytics.bigdl.tensor.{Storage, Tensor}
-import org.apache.spark.ml.param.{ParamMap, _}
+import com.intel.analytics.bigdl.{Criterion, Module}
+import org.apache.spark.ml.param.ParamMap
+import org.apache.spark.ml.util.SchemaUtils
 import org.apache.spark.sql.types._
-import org.apache.spark.sql.{DataFrame, Row}
-
-// TODO: override transformSchema to change label and prediction type
-class DLClassifier[@specialized(Float, Double) T: ClassTag] (
-    override val model: Module[T],
-    featureSize : Array[Int],
-    override val uid: String = "DLClassifier"
-  ) (implicit ev: TensorNumeric[T]) extends DLModel[T](model, featureSize) {
 
-}
+import scala.reflect.ClassTag
 
 /**
- * [[DLModel]] helps embedding a BigDL model into a Spark Transformer, thus Spark users can
- * conveniently merge BigDL into Spark ML pipeline. The features column holds the storage
- * (Vector, float array or double array) of the feature data, and user should specify the
- * tensor size (dimensions) via featureSize. (e.g. an image may be with featureSize = 28 * 28).
+ * [[DLClassifier]] is a specialized [[DLEstimator]] that simplifies the data format for
+ * classification tasks. It only supports label column of DoubleType.
+ * and the fitted [[DLClassifierModel]] will have the prediction column of DoubleType.
  *
- * Internally the feature data are converted to BigDL tensors with batch acceleration, and
- * further predict with a BigDL model.
- *
- * [[DLModel]] is compatible with both spark 1.5-plus and 2.0 by extending ML Transformer.
+ * @param model BigDL module to be optimized
+ * @param criterion  BigDL criterion method
+ * @param featureSize The size (Tensor dimensions) of the feature data.
  */
-class DLModel[@specialized(Float, Double) T: ClassTag](
-    val model: Module[T],
-    var featureSize : Array[Int],
-    override val uid: String = "DLModel"
-  )(implicit ev: TensorNumeric[T]) extends DLTransformerBase with DLParams with HasBatchSize {
-
-  def setFeaturesCol(featuresColName: String): this.type = set(featuresCol, featuresColName)
-
-  def setPredictionCol(value: String): this.type = set(predictionCol, value)
+class DLClassifier[@specialized(Float, Double) T: ClassTag](
+    override val model: Module[T],
+    override val criterion : Criterion[T],
+    override val featureSize : Array[Int],
+    override val uid: String = "DLClassifier"
+  )(implicit ev: TensorNumeric[T])
+  extends DLEstimator[T](model, criterion, featureSize, Array(1)) with DLParams with HasBatchSize {
```

Comment:
Do we need to explicitly add "with DLParams with HasBatchSize" as general class DLEstimator already does ?